oauth2_http/javatests/com/google/auth/oauth2/ServiceAccountCredentialsTest.java

```diff
@@ -975,6 +976,58 @@ public void idTokenWithAudience_iamFlow_targetAudienceDoesNotMatchAudClaim() thr
         tokenCredential.getIdToken().getJsonWebSignature().getPayload().getAudience());
   }
 
+  @Test
+  public void idTokenWithAudience_oauthEndpoint_non2XXStatusCode() throws IOException {
+    String universeDomain = "test.com";
+    MockTokenServerTransportFactory transportFactory = new MockTokenServerTransportFactory();
+    transportFactory.transport.setError(new IOException("404 Not Found"));
+    ServiceAccountCredentials credentials =
+        createDefaultBuilder()
+            .setScopes(SCOPES)
+            .setHttpTransportFactory(transportFactory)
+            .setUniverseDomain(universeDomain)
+            .build();
+
+    String targetAudience = "audience";
+    IdTokenCredentials tokenCredential =
+        IdTokenCredentials.newBuilder()
+            .setIdTokenProvider(credentials)
+            .setTargetAudience(targetAudience)
+            .build();
+
+    // Ensure that a non 2xx status code returns an exception and doesn't continue execution
+    assertThrows(IOException.class, tokenCredential::refresh);
+  }
+
+  @Test
+  public void idTokenWithAudience_iamEndpoint_non2XXStatusCode() throws IOException {
+    String universeDomain = "test.com";
```

Comment:
same here

Reply:
Non GDU will hit the IAM Endpoint. Setting the Universe Domain is needed here